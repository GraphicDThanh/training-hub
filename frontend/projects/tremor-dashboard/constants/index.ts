export * from "./account";
export * from "./analytics";
export * from "./api";
export * from "./billing";
export * from "./breakpoints";
export * from "./colors";
export * from "./common";
export * from "./flagSrc";
export * from "./images";
export * from "./pagination";
export * from "./messages";
export * from "./pincode";
export * from "./products";
export * from "./projects";
export * from "./profile";
export * from "./regex";
export * from "./routes";
export * from "./sales";
export * from "./saleStatistical";
export * from "./sections";
export * from "./status";
export * from "./units";
export * from "./keyboard";
export * from "./steps";
export * from "./users";
export * from "./conversation";
