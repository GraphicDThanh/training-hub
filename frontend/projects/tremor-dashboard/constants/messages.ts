export const MESSAGES_ERROR = {
  NAME_MIN_LENGTH: "Minimum length should be 4.",
  NAME_INVALID: "This field is no space at start or end",
  NAME_REQUIRED: "Name is required.",
  PASSWORD_REQUIRED: "Password is required.",
  EMAIL_REQUIRED: "Email is required.",
  EMAIL_INVALID: "Invalid email address.",
  SKU_INVALID: "Invalid SKU number.",
  PASSWORD_INVALID:
    "Password must have at least 8 characters, 1 special character and 1 number.",
  FIELD_REQUIRED: "This field is required.",
  MIN_LENGTH_4: "This field must have more than 4 characters.",
  URL_REQUIRED: "URL is required.",
  INVALID_URL: "URL is invalid.",
  INVALID_PHONE_NUMBER: "Phone number is invalid",
  INVALID_BIRTH_DATE: "Birth date is invalid.",
  REPEAT_PASSWORD_INVALID: "Password not matching",
  TERMS_REQUIRED: "Please agree with the terms and conditions to continue.",
  CONFIRM_EMAIL: "The email confirmation does not match.",
  PRODUCT: {
    WEIGHT_LIMIT: "Weight has limit is 2000 pound.",
    QUANTITY_LIMIT: "Quantity has limit is 10000.",
    PRICE_LIMIT: "Price has limit is $20,000.",
  },
  TRANSACTION: {
    AMOUNT_LIMIT: "Amount has limit is $20,000.",
    MIN_AMOUNT: "Amount must be positive number.",
  },
};

export const MESSAGES_WARNING = {
  PRODUCT: {
    WEIGHT_IS_ZERO: "Missing weight is not encourage.",
    QUANTITY_IS_ZERO:
      "This product will has label sold out if have quantity default is 0.",
    ACCEPT_IMAGE_UPLOAD:
      "(We accept .png, .jpg, .jpeg, and .webp files, up to 32MB in size.)",
  },
};

export const MESSAGE_SIGN_UP = {
  SUCCESS: "Sign up successfully.",
  FAILED: "Failed to sign up",
};

export const MESSAGE_SIGN_IN = {
  FAILED: "Failed to sign in",
  SUCCESS: "Sign in successfully.",
  CREDENTIALS: "Please make sure your email address and password are correct.",
};

export const MESSAGE_ADD_PRODUCT = {
  SUCCESS: "Add product successfully",
  FAILED: "Failed to add product",
};

export const MESSAGE_EDIT_PRODUCT = {
  SUCCESS: "Edit product successfully",
  FAILED: "Failed to edit product",
};

export const MESSAGE_USER = {
  MAIL_EXISTS: "Account with this email already exists!",
  ADD_FAILED: "Failed to create new account. Please try again!",
  EDIT_SUCCESS: "Edit user successfully",
  EDIT_FAILED: "Failed to edit user",
  GET_USER_FAILED: "Failed to get user",
};

export const MESSAGE_SIGN_OUT = {
  SUCCESS: "Sign out successfully.",
  FAILED: "Failed to Sign out",
};

export const MESSAGE_ADD_PROJECT = {
  SUCCESS: "Add project successfully",
  FAILED: "Failed to add project",
};

export const MESSAGE_EDIT_PROJECT = {
  SUCCESS: "Edit project successfully",
  FAILED: "Failed to edit project",
};

export const MESSAGE_DELETE_PROJECT = {
  SUCCESS: "Delete project successfully",
  FAILED: "Failed to delete project",
};

export const MESSAGE_ADD_USER = {
  SUCCESS: "Add new user successfully",
  FAILED: "Failed to add new user",
};

export const MESSAGES_TRANSACTION = {
  ADD_SUCCESS: "Create new transaction successfully",
  ADD_FAILED: "Failed to create new transaction. Please try again!",
};
