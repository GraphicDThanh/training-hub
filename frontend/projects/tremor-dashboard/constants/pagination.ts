export const PAGE_SIZE = {
  SIZE: 10,
};
