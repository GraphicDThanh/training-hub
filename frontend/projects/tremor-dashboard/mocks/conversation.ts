import * as ConversationContext from "@/context/conversation";
import { ConversationGroupExtend } from "@/hooks/useConversationGroups";

import { MOCKS_CHAT_BOX, MOCKS_CONVERSATION_GROUP } from "@/mocks";

const renderMockId = () => Math.floor(Math.random() * 100000000).toString();

export const renderChatData = () => ({
  data: () => MOCKS_CHAT_BOX[Math.abs(Math.floor(Math.random() * 10) - 5)],
  id: renderMockId(),
  exists: () => true,
});

export const renderGroupData = () => ({
  data: () =>
    MOCKS_CONVERSATION_GROUP[Math.abs(Math.floor(Math.random() * 10) - 5)],
  id: renderMockId(),
  exists: () => true,
});

export const mockUseConversation = (group = {} as ConversationGroupExtend) => {
  jest.spyOn(ConversationContext, "useConversation").mockImplementation(() => ({
    onOpenConversationModal: () => {},
    setGroup: () => {},
    group,
  }));
};
