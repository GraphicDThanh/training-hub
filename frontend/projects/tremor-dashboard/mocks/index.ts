export * from "./analytics";
export * from "./billing";
export * from "./invoices";
export * from "./navigation";
export * from "./orders";
export * from "./products";
export * from "./profile";
export * from "./projects";
export * from "./sales";
export * from "./tableList";
export * from "./transactions";
export * from "./users";
export * from "./images";
export * from "./firebase";
export * from "./conversation";
export * from "./session";
