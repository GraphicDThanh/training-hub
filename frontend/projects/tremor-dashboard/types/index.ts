export * from "./analytics";
export * from "./api";
export * from "./billing";
export * from "./breadcrumb";
export * from "./common";
export * from "./invoice";
export * from "./navigation";
export * from "./orderDetails";
export * from "./orders";
export * from "./product";
export * from "./profile";
export * from "./project";
export * from "./sales";
export * from "./select";
export * from "./table-list";
export * from "./table";
export * from "./transactions";
export * from "./user";
