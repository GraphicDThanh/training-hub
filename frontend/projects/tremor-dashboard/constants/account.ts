// Types
import { OptionType } from "@/types";

export const GENDERS: OptionType[] = [
  {
    option: "Male",
    value: "0",
  },
  {
    option: "Female",
    value: "1",
  },
];

export const SKILLS: Record<string, string | number>[] = [
  {
    value: "0",
    option: "React",
  },
  {
    value: "1",
    option: "Vue",
  },
  {
    value: "2",
    option: "Angular",
  },
  {
    value: "3",
    option: "Svelte",
  },
  {
    value: "4",
    option: "Javascript",
  },
];

export const MONTHS: OptionType[] = [
  {
    option: "January",
    value: "0",
  },
  {
    option: "February",
    value: "1",
  },
  {
    option: "March",
    value: "2",
  },
  {
    option: "April",
    value: "3",
  },
  {
    option: "May",
    value: "4",
  },
  {
    option: "June",
    value: "5",
  },
  {
    option: "July",
    value: "6",
  },
  {
    option: "August",
    value: "7",
  },
  {
    option: "September",
    value: "8",
  },
  {
    option: "October",
    value: "9",
  },
  {
    option: "November",
    value: "10",
  },
  {
    option: "December",
    value: "11",
  },
];

export const ROLES = [
  {
    value: 0,
    label: "User",
  },
  {
    value: 1,
    label: "Admin",
  },
];
