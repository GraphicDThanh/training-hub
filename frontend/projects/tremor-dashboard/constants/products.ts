// Types
import { OptionType, TFieldSocial } from "@/types";

type TopSellingProductsType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
  5: string;
};

export const TOP_SELLING_PRODUCTS_IMAGE: TopSellingProductsType = {
  1: "/assets/images/products/blue-shoe.webp",
  2: "/assets/images/products/black-mug.webp",
  3: "/assets/images/products/black-chair.webp",
  4: "/assets/images/products/bang-sound.webp",
  5: "/assets/images/products/photo-tools.webp",
};

export const PRICE_TYPE: OptionType[] = [
  {
    option: "BTC",
    value: "1",
  },
  {
    option: "CNY",
    value: "2",
  },
  {
    option: "EUR",
    value: "3",
  },
  {
    option: "GBP",
    value: "4",
  },
  {
    option: "INR",
    value: "5",
  },
  {
    option: "USD",
    value: "6",
  },
];

export const PRICE_TAGS: OptionType[] = [
  {
    option: "Black Friday",
    value: "1",
  },
  {
    option: "Expired",
    value: "2",
  },
  {
    option: "Sale",
    value: "3",
  },
];

export const PRODUCT_CATEGORY: OptionType[] = [
  {
    option: "Clothing",
    value: "1",
  },
  {
    option: "Electronics",
    value: "2",
  },
  {
    option: "Furniture",
    value: "3",
  },
  {
    option: "Others",
    value: "4",
  },
  {
    option: "Real Estate",
    value: "5",
  },
];

export const OTHER_PRODUCTS_LENGTH = 4;

export const SOCIAL_PRODUCTS: TFieldSocial[] = [
  {
    id: "shopifyUrl",
    name: "shopifyUrl",
    label: "Shopify Handle",
    value: "",
  },
  {
    id: "facebookUrl",
    name: "facebookUrl",
    label: "Facebook Account",
    value: "",
  },
  {
    id: "instagramUrl",
    name: "instagramUrl",
    label: "Instagram Account",
    value: "",
  },
];
