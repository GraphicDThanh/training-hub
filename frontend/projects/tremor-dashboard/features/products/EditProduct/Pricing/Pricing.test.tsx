import {
  fireEvent,
  render,
  queryAllByAttribute,
  waitFor,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import { FormProvider, useForm } from "react-hook-form";

// Components
import PricingInfo from "./Pricing";

// Constants
import { MESSAGES_ERROR, EXCEPT_KEYS } from "@/constants";

// Types
import { EditProductData } from "@/types";

// Mocks
import { mockProductInfo } from "@/mocks";

interface ProductInfoFormProps {
  price: number;
}

const PricingInfoForm = ({ price = 0 }: ProductInfoFormProps) => {
  const formHandler = useForm<EditProductData>({
    defaultValues: {
      price,
    },
    mode: "onBlur",
  });
  return (
    <FormProvider {...formHandler}>
      <PricingInfo />
    </FormProvider>
  );
};

describe("Testing PricingInfo component", () => {
  const renderComponent = () =>
    render(<PricingInfoForm {...mockProductInfo} />);

  it("Should match snapshot", () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  it("Should not show error messages for valid price", async () => {
    const validPrice = 123;
    const { queryByText, getByTestId } = renderComponent();
    const inputPrice = getByTestId("edit-price");

    fireEvent.change(inputPrice, { target: { value: validPrice } });
    fireEvent.blur(inputPrice);
    expect(inputPrice.value).toEqual("123");
    expect(queryByText(MESSAGES_ERROR.FIELD_REQUIRED)).toBeNull();
  });

  it("Should show error messages for limit price", async () => {
    const limitPrice = 22222;

    const { queryByText, getByTestId } = renderComponent();
    const inputPrice = getByTestId("edit-price");

    fireEvent.change(inputPrice, { target: { value: limitPrice } });
    fireEvent.blur(inputPrice);
    expect(inputPrice.value).toEqual("22,222");
    await waitFor(() =>
      expect(queryByText(MESSAGES_ERROR.PRODUCT.PRICE_LIMIT)).toBeTruthy(),
    );
  });

  it("Should not show error messages for valid price", async () => {
    const validPrice = 123;
    const { queryByText, getByTestId } = renderComponent();
    const inputPrice = getByTestId("edit-price");

    fireEvent.change(inputPrice, { target: { value: validPrice } });
    fireEvent.blur(inputPrice);
    expect(inputPrice.value).toEqual("123");
    expect(queryByText(MESSAGES_ERROR.FIELD_REQUIRED)).toBeNull();
  });

  it("Should show error messages for empty price", async () => {
    const emptyPrice = "";

    const { queryByText, getByTestId } = renderComponent();
    const inputPrice = getByTestId("edit-price");

    fireEvent.change(inputPrice, { target: { value: emptyPrice } });
    fireEvent.blur(inputPrice);
    expect(inputPrice.value).toEqual("");

    await waitFor(() =>
      expect(queryByText(MESSAGES_ERROR.FIELD_REQUIRED)).toBeTruthy(),
    );
  });

  it("Should not show error messages for valid SKU", async () => {
    const validSKU = 123;

    const { queryByText, getByTestId } = renderComponent();
    const inputSku = getByTestId("edit-sku");

    fireEvent.change(inputSku, { target: { value: validSKU } });
    fireEvent.blur(inputSku);
    expect(inputSku.value).toEqual("123");

    await waitFor(() =>
      expect(queryByText(MESSAGES_ERROR.SKU_INVALID)).not.toBeTruthy(),
    );
  });

  it("Should show error messages for invalid SKU", async () => {
    const invalidSKU = "invalid";
    const { queryByText, getByTestId } = renderComponent();
    const inputSku = getByTestId("edit-sku");

    fireEvent.change(inputSku, { target: { value: invalidSKU } });
    fireEvent.blur(inputSku);
    expect(inputSku.value).toEqual("invalid");

    await waitFor(() =>
      expect(queryByText(MESSAGES_ERROR.SKU_INVALID)).toBeTruthy(),
    );
  });

  it("Should call preventDefault when key is in except key", async () => {
    const mockEvent = {
      key: EXCEPT_KEYS.POSITIVE_DOUBLE[3],
    };
    const getById = queryAllByAttribute.bind(null, "id");
    const { container } = renderComponent();
    const priceInput = getById(container, "edit-price");

    fireEvent.keyDown(priceInput[0], mockEvent); // Invalid input

    expect(priceInput[0]).toBeInTheDocument();
  });
});
