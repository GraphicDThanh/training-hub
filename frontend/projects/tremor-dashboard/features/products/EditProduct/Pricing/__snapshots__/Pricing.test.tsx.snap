// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PricingInfo component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default pricing-info"
      >
        <p
          class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-6"
        >
          Pricing
        </p>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-end"
        >
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-start sm:flex-row"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center"
            >
              <div
                class="tremor-Flex-root flex w-full flex-col justify-between items-center sm:flex-row"
              >
                <div
                  class="w-full min-h-20"
                >
                  <div
                    class="w-full"
                  >
                    <label
                      class="text-grey text-sm dark:text-lighter after:content-['*'] after:text-red-500 after:text-md"
                      for="edit-price"
                    >
                      Price
                    </label>
                    <div
                      class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
                    >
                      <span
                        class="absolute top-0 left-0 text-sm min-h-[30px] py-1.5 text-primary dark:text-white"
                      >
                        $
                      </span>
                      <div
                        class="w-full"
                      >
                        <input
                          class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 pl-[15px]"
                          data-testid="edit-price"
                          id="edit-price"
                          placeholder="0,00"
                          type="text"
                          value="0"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="w-full sm:ml-6 min-h-20"
                >
                  <div
                    class="w-full"
                  >
                    <label
                      class="text-grey text-sm dark:text-lighter"
                      for="edit-sku"
                    >
                      SKU
                    </label>
                    <div
                      class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
                    >
                      <div
                        class="w-full"
                      >
                        <input
                          class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 "
                          data-testid="edit-sku"
                          id="edit-sku"
                          name="sku"
                          type="text"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="w-full min-h-20"
              >
                <label
                  class="text-gray-500 text-sm dark:text-gray-400"
                  for="tags"
                >
                  Tags
                </label>
                <div
                  class="w-full min-w-[10rem] relative text-tremor-default"
                >
                  <select
                    class="h-full w-full absolute left-0 top-0 z-0 opacity-0"
                    multiple=""
                    title="multi-select-hidden"
                  >
                    <option
                      class="hidden"
                      disabled=""
                      hidden=""
                      value=""
                    >
                      Select...
                    </option>
                    <option
                      class="hidden"
                      value="1"
                    >
                      Black Friday
                    </option>
                    <option
                      class="hidden"
                      value="2"
                    >
                      Expired
                    </option>
                    <option
                      class="hidden"
                      value="3"
                    >
                      Sale
                    </option>
                  </select>
                  <div
                    class="w-full min-w-[10rem] relative text-tremor-default select-custom select-multi dark:text-white dark:border-light dark:focus:border-white"
                    data-headlessui-state=""
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 border pr-8 py-1.5 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-:r0:"
                      type="button"
                    >
                      <div
                        class="h-6 flex items-center"
                      >
                        <span>
                          Select...
                        </span>
                      </div>
                      <span
                        class="absolute inset-y-0 right-0 flex items-center mr-2.5"
                      >
                        <svg
                          class="tremor-MultiSelect-arrowDownIcon flex-none h-5 w-5 text-tremor-content-subtle dark:text-dark-tremor-content-subtle"
                          fill="currentColor"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11.9999 13.1714L16.9497 8.22168L18.3639 9.63589L11.9999 15.9999L5.63599 9.63589L7.0502 8.22168L11.9999 13.1714Z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default pricing-info"
    >
      <p
        class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-6"
      >
        Pricing
      </p>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-between items-end"
      >
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-start sm:flex-row"
        >
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-center"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center sm:flex-row"
            >
              <div
                class="w-full min-h-20"
              >
                <div
                  class="w-full"
                >
                  <label
                    class="text-grey text-sm dark:text-lighter after:content-['*'] after:text-red-500 after:text-md"
                    for="edit-price"
                  >
                    Price
                  </label>
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
                  >
                    <span
                      class="absolute top-0 left-0 text-sm min-h-[30px] py-1.5 text-primary dark:text-white"
                    >
                      $
                    </span>
                    <div
                      class="w-full"
                    >
                      <input
                        class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 pl-[15px]"
                        data-testid="edit-price"
                        id="edit-price"
                        placeholder="0,00"
                        type="text"
                        value="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="w-full sm:ml-6 min-h-20"
              >
                <div
                  class="w-full"
                >
                  <label
                    class="text-grey text-sm dark:text-lighter"
                    for="edit-sku"
                  >
                    SKU
                  </label>
                  <div
                    class="tremor-Flex-root flex w-full flex-row justify-between items-center relative"
                  >
                    <div
                      class="w-full"
                    >
                      <input
                        class="block py-1.5 px-0 w-full text-sm text-primary bg-transparent border-0 border-b-2 border-gray-300 appearance-none dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer disabled:cursor-not-allowed disabled:opacity-50 "
                        data-testid="edit-sku"
                        id="edit-sku"
                        name="sku"
                        type="text"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="w-full min-h-20"
            >
              <label
                class="text-gray-500 text-sm dark:text-gray-400"
                for="tags"
              >
                Tags
              </label>
              <div
                class="w-full min-w-[10rem] relative text-tremor-default"
              >
                <select
                  class="h-full w-full absolute left-0 top-0 z-0 opacity-0"
                  multiple=""
                  title="multi-select-hidden"
                >
                  <option
                    class="hidden"
                    disabled=""
                    hidden=""
                    value=""
                  >
                    Select...
                  </option>
                  <option
                    class="hidden"
                    value="1"
                  >
                    Black Friday
                  </option>
                  <option
                    class="hidden"
                    value="2"
                  >
                    Expired
                  </option>
                  <option
                    class="hidden"
                    value="3"
                  >
                    Sale
                  </option>
                </select>
                <div
                  class="w-full min-w-[10rem] relative text-tremor-default select-custom select-multi dark:text-white dark:border-light dark:focus:border-white"
                  data-headlessui-state=""
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 border pr-8 py-1.5 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r0:"
                    type="button"
                  >
                    <div
                      class="h-6 flex items-center"
                    >
                      <span>
                        Select...
                      </span>
                    </div>
                    <span
                      class="absolute inset-y-0 right-0 flex items-center mr-2.5"
                    >
                      <svg
                        class="tremor-MultiSelect-arrowDownIcon flex-none h-5 w-5 text-tremor-content-subtle dark:text-dark-tremor-content-subtle"
                        fill="currentColor"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M11.9999 13.1714L16.9497 8.22168L18.3639 9.63589L11.9999 15.9999L5.63599 9.63589L7.0502 8.22168L11.9999 13.1714Z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
